nx/lib/nx/defn/expr.ex

```diff
@@ -213,7 +213,14 @@ defmodule Nx.Defn.Expr do
   """
   def while(initial, context, arg, condition, body) do
     [flatten_initial, flatten_arg, flatten_body] = clauses = flatten_clauses([initial, arg, body])
```

Comment:
I am thinking we should no longer change `reshape_vectors` and friend to work on containers and instead move the devectorize to flatten_clauses.